projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -61,7 +73,16 @@ public static function build_view_data() {
 		} else {
 			$post = '0';
 		}
-		return compact( 'v', 'blog', 'post', 'tz', 'srv' );
+		$view_data = compact( 'v', 'blog', 'post', 'tz', 'srv' );
+		foreach ( self::TRACKED_UTM_PARAMETERS as $utm_parameter ) {
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			if ( isset( $_GET[ $utm_parameter ] ) && is_scalar( $_GET[ $utm_parameter ] ) ) {
+				// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+				$view_data[ $utm_parameter ] = substr( (string) $_GET[ $utm_parameter ], 0, 255 );
```

Comment:
What would you think about adding some of the recommended sanitization methods here? UTM parameter values are limited on Google's end as well I believe, so why not do the same on our end? It would also help avoid issues like PHP notices being logged when someone tries to do something like `?utm_source=<script>alert( 'hello' )</script>`.

```
PHP Notice:  Function wp_add_inline_script was called incorrectly. Do not pass &lt;script&gt; tags to wp_add_inline_script(). Please see <a>Debugging in WordPress</a> for more information. (This message was added in version 4.5.0.) in /wp-includes/functions.php on line 6031
```

```suggestion
				// phpcs:ignore WordPress.Security.NonceVerification.Recommended -- We're sanitizing the value.
				$view_data[ $utm_parameter ] = substr( sanitize_textarea_field( wp_unslash( $_GET[ $utm_parameter ] ) ), 0, 255 );
```

Reply:
I've added the sanitization. It does limit the valid UTM parameters we can handle slightly, but it shouldn't come up in a real-world scenarios too often.

By the way, would you know why `Tracking_Pixel::stats_array_to_string` didn't use `json_encode` directly? I've changed it now that you've highlighted the importance of escaping the data properly. I don't think `addslashes` was sufficient there, and it wasn't checking keys, which could have been overlooked when adding more parameters.

Reply:
> would you know why Tracking_Pixel::stats_array_to_string didn't use json_encode directly?

Just a fun guess because I was curious... json_encode() was added in php5.2 in late 2006. The original stats_array() function on wpcom in mu-plugins was written in Feb 2007!

Reply:
Oh, thanks for checking! I didn't realize it was moved from an old code, that would definitely explain it. I think it's worth changing it to json_encode then, just in case we change array keys in the future.